pkg/plugin/run.go

```diff
@@ -163,12 +169,26 @@ func (e *AutoscaleEnforcer) handleAgentRequest(
 		return nil, 400, errors.New("pod is not associated with a VM")
 	}
 
+	// Check that req.ComputeUnit.Mem is divisible by the VM's memory slot size
+	if req.ComputeUnit != nil && req.ComputeUnit.Mem%pod.vm.memSlotSize != 0 {
+		return nil, 400, fmt.Errorf(
+			"computeUnit is not divisible by VM memory slot size: %v not divisible by %v",
+			req.ComputeUnit,
+			pod.vm.memSlotSize,
+		)
+	}
+
 	// If the request was actually sending a quantity of *memory slots*, rather than bytes, then
 	// multiply memory resources to make it match the
 	if !req.ProtoVersion.RepresentsMemoryAsBytes() {
 		req.Resources.Mem *= pod.vm.memSlotSize
 	}
 
+	computeUnit := e.state.conf.ComputeUnit
```

Comment:
I thought the plan is to remove `ComputeUnit` from scheduler's config.

Reply:
That's the plan, yeah - see #743. But the intermediate states should be stable, in particular any release must fully support new scheduler + old autoscaler-agent. IMO it's probably too complicated to try to do that if we merge #653, this PR, _and_ #743, so I figured that PR should be delayed until the release after. But we _do_ want this PR in the release with #653 in order to be able to accurately use the compute unit when handling agent requests.